### PR TITLE
Use "@id" as the data type for IRIs

### PR DESCRIPTION
**Pull request by zonotope**
This patch uses "@id" as the internal datatype for IRI values. It builds off of #860

### CODE REVIEW

**Review by bplatz — APPROVED** [review]
I was thinking this would affect the `values` for policy enforcement but looks like you addressed that as well.

Nice to have this evolution of how we think of 'node' identifiers without needing an explicit scalar datatypes.

File: src/clj/fluree/db/json_ld/policy/enforce.cljc [hunk 1]
```diff
@@ -55,7 +55,7 @@
                    :TODO
                    [(into ["?$this"] (keys values-map))
                     [(into [{"@value" this-var
-                             "@type"  "http://www.w3.org/2001/XMLSchema#anyURI"}] (vals values-map))]])]
+                             "@type"  "@id"}] (vals values-map))]])]
```
Comment (dpetran) [inline src/clj/fluree/db/json_ld/policy/enforce.cljc:58]: I think we should use `const/iri-id` here.

**Review by dpetran — APPROVED** [review]
:+1:

**Comment by zonotope** [comment]
I'm holding off on merging this because it needs a migration for existing databases. I am still working on some indexing stuff, but I'll return to the migration when I'm done with that.

**Pull request merged**